src/engine/build.sh

```diff
@@ -247,18 +250,19 @@ case $B2_TOOLSET in
     ;;
 
     intel-linux)
-        CXX=${CXX:=icc}
+        CXX=${CXX:=icpc}
         CXX_VERSION_OPT=${CXX_VERSION_OPT:=--version}
-        test_path ${CXX} >/dev/null 2>&1
-        if test $? ; then
+        if test_path ${CXX} ; then
             echo "Found ${CXX} in environment"
-            B2_TOOLSET_ROOT=`echo ${CXX}| sed -e 's/bin.*\/icc//'`
+            B2_TOOLSET_ROOT=`echo ${CXX}| sed -e 's/bin.*\/icpc//'`
             # probably the most widespread
             ARCH=intel64
         else
             echo "No intel compiler in current path"
-            echo "Look in a few old place for legacy reason"
-            if test -r /opt/intel/cc/9.0/bin/iccvars.sh ; then
+            echo "Look in a few common places just in case"
+	    if test -r /opt/intel/inteloneapi/setvars.sh ; then
```

Comment:
Just a minor formatting problem:
```suggestion
            if test -r /opt/intel/inteloneapi/setvars.sh ; then
```